op-wheel/cheat/cheat.go

```diff
@@ -81,7 +81,12 @@ func (ch *Cheater) Close() error {
 func openStorageTrie(s *state.StateDB, addr common.Address) (state.Trie, error) {
 	stateRoot := s.IntermediateRoot(true)
 	storageRoot := s.GetStorageRoot(addr)
-	return s.Database().OpenStorageTrie(stateRoot, addr, storageRoot)
+	storageTrie, err := s.Database().OpenTrie(storageRoot)
+	if err != nil {
+		return nil, fmt.Errorf("opening storage trie: %w", err)
+	}
+	// TODO(Seb) confirm
+	return s.Database().OpenStorageTrie(stateRoot, addr, storageRoot, storageTrie)
```

Comment:
`OpenStorageTrie` now receives an additional `self Trie` as its last parameter that is returned directly in a Verkle tree case, AFAIU from the [internal cachingDB implementation](https://github.com/ethereum/go-ethereum/blob/566754c74a74c8175ec2f1ee5cc10a8caced6015/core/state/database.go#L191-L198).

Reviewers please confirm that this makes sense here. Can alternatively also pass `nil` in non-Verkle case.